api/models/User.js

```diff
@@ -170,43 +171,50 @@ module.exports = bookshelf.Model.extend(merge({
     return this.save({ active: false })
   },
 
+  deleteUserMedia: async function () {
+    const userId = this.get('id')
+    const userUrls = [this.get('banner_url'), this.get('avatar_url')]
+
+    const mediaUrls = await Media.findMediaUrlsForUser(userId)
+    const urls = mediaUrls.concat(userUrls)
+    Queue.classMethod('Media', 'deleteMediaByUrl', { urls })
+  },
+
   sanelyDeleteUser: async function ({ sessionId, transacting = {} }) {
-    const deletedUserResult = await bookshelf.knex.raw('select id from users where email = \'deleted@hylo.com\'')
-    const deletedUserId = deletedUserResult.rows[0].id
-    Queue.classMethod('User', 'clearSessionsFor', { userId: this.get('user_id'), sessionId })
     /* 
       ### List of things to be done on account deletion ###
 
+      - Look up urls for all their possible uploads
+      - drop urls from external sources
+      - iterate through urls to delete each upload
+
       - zero out content of their posts and comments
-      - change the ownership of their posts and comments to the deletedUser record
       - remove other references to their user_id
-      - delete their user record!
+      - wipe their user record!
     */
 
+    await this.deleteUserMedia()
+    Queue.classMethod('User', 'clearSessionsFor', { userId: this.get('user_id'), sessionId })
+
     const query = `
     BEGIN;
     UPDATE posts SET name = 'Post by deleted user', description = '', location = NULL, location_id = NULL WHERE user_id = ${this.id};
-    UPDATE posts_users SET user_id = ${deletedUserId} WHERE user_id = ${this.id};
     DELETE FROM user_connections WHERE (user_id = ${this.id}) OR (other_user_id = ${this.id});
 
-    UPDATE groups SET created_by_id = ${deletedUserId} WHERE created_by_id = ${this.id};
-    UPDATE comments SET deactivated_by_id = ${deletedUserId} WHERE deactivated_by_id = ${this.id};
     UPDATE comments SET text = 'Comment by deleted user' WHERE user_id = ${this.id};
 
-    UPDATE follows set added_by_id = ${deletedUserId} WHERE added_by_id = ${this.id};
-    UPDATE groups_tags set user_id = ${deletedUserId} WHERE user_id = ${this.id};
     DELETE FROM thanks WHERE comment_id in (select id from comments WHERE user_id = ${this.id});
     DELETE FROM thanks WHERE thanked_by_id = ${this.id};
     DELETE FROM notifications WHERE activity_id in (select id from activities WHERE reader_id = ${this.id});
     DELETE FROM notifications WHERE activity_id in (select id from activities WHERE actor_id = ${this.id});
+    DELETE FROM notifications WHERE activity_id in (select id from activities WHERE parent_comment_id in (select id from comments where user_id = ${this.id}));
```

Comment:
so now that im thinking about this, won't this delete notifications for other people that have comments on this person's comments? I think we need to check that the notification or the activity also is for this particular user? im a bit confused...

Reply:
I think this was added to ensure its counterpart of

`DELETE from activities WHERE parent_comment_id in (select id from comments where user_id = ${this.id});` would work without hitting a constraint.

However, I wonder now whether this `DELETE from` is even necessary and whether they can both be removed?

Reply:
I think we should talk this through